relay-server/src/actors/processor.rs

```diff
@@ -541,6 +557,7 @@ impl EnvelopeProcessorService {
         project_cache: Addr<ProjectCache>,
         upstream_relay: Addr<UpstreamRelay>,
     ) -> Self {
+        let global_config = RwLock::new(Arc::new(GlobalConfig::default()));
```

Comment:
For now, we can treat the `GlobalConfig` as optional, but I agree with @olksdr's earlier comments here:
```suggestion
        let global_config = RwLock::new(Arc::new(GlobalConfig::default()));
```

https://github.com/getsentry/relay/pull/2320#discussion_r1267751746

```suggestion
        // TODO: Replace with state enum (`init`, `ready`) and do not process anything while global_config is undefined.
        let global_config = RwLock::new(Arc::new(GlobalConfig::default()));
```

Reply:
yep! coming in next PR